src/dynsight/_internal/vision/detect.py

```diff
@@ -457,65 +496,132 @@ def fit(
                                 "confidence": float(conf[i]),
                             }
                         )
-            # Look for outliers in the boxes width and height
-            widths = np.array(
-                [d["width"] for d in detection_results], dtype=float
+            logger.info("Looking for outliers")
+            # Filter detections
+            detection_results = self._filter_detections(
+                detection_results,
+                prediction_number,
             )
-            heights = np.array(
-                [d["height"] for d in detection_results],
-                dtype=float,
+            logger.info(f"Building the dataset (version {prediction_number}")
+            # Build the new dataset
+            self._build_dataset(
+                detection_results=detection_results,
+                dataset_name=f"dataset_{prediction_number}",
             )
-            outliers_plt_folder = (
-                self.project_folder
-                / "predictions"
-                / f"attempt_{prediction_number}"
-                / "outliers"
+            # Remove the oldest dataset in config
+            # It has been made to avoid training bias on worst results
+            self._remove_old_dataset()
+
+            train_dataset_path = (
+                self._project_folder
+                / "train_datasets"
+                / f"dataset_{prediction_number}"
             )
-            outliers_plt_folder.mkdir(exist_ok=True)
+            # Update the dataset config file
+            self._add_or_create_yaml(train_dataset_path)
+
+    def compute_trj(
```

Comment:
Can we change this method name? I do not know what it means to `compute_trj`. Also, in dynsight, `trj` now means something specific.